### PR TITLE
mcs: tso server compare address without scheme

### LINKED ISSUE #8284: mcs: tso servers cannot election after replacing `http` with `https`
### What did you do?

Replace `http` with `https` in tso server scheme

### What did you expect to see?

Work

### What did you see instead?

tso server cannot election and always resign election membership
image

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #8284

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/keyspace/tso_keyspace_group.go [hunk 1]
```diff
@@ -874,7 +875,7 @@ func (m *GroupManager) SetPriorityForKeyspaceGroup(id uint32, node string, prior
 		inKeyspaceGroup := false
 		members := make([]endpoint.KeyspaceGroupMember, 0, len(kg.Members))
 		for _, member := range kg.Members {
-			if member.Address == node {
+			if typeutil.CompareURLsWithoutScheme(member.Address, node) {
```
Comment (rleungx) [inline pkg/keyspace/tso_keyspace_group.go:878]: Please add a comment to explain why we do it.

File: pkg/utils/typeutil/comparison.go [hunk 2]
```diff
@@ -78,3 +79,17 @@ func AreStringSlicesEquivalent(a, b []string) bool {
...
+// CompareURLsWithoutScheme compares two URLs without scheme.
+func CompareURLsWithoutScheme(url1, url2 string) bool {
+	trimScheme := func(url string) string {
+		if strings.HasPrefix(url, "http://") {
+			return strings.TrimPrefix(url, "http://")
+		} else if strings.HasPrefix(url, "https://") {
+			return strings.TrimPrefix(url, "https://")
+		}
+		return url
+	}
```
Comment (iosmanthus) [inline pkg/utils/typeutil/comparison.go:92]:
```
	trimScheme := func(s string) string {
		return strings.TrimPrefix(strings.TrimPrefix(s, "https://"), "http://")
	}
```

File: pkg/utils/typeutil/comparison.go [hunk 3]
```diff
@@ -78,3 +79,17 @@ func AreStringSlicesEquivalent(a, b []string) bool {
 func Float64Equal(a, b float64) bool {
 	return math.Abs(a-b) <= 1e-6
 }
+
+// CompareURLsWithoutScheme compares two URLs without scheme.
+func CompareURLsWithoutScheme(url1, url2 string) bool {
```
Comment (iosmanthus) [inline pkg/utils/typeutil/comparison.go:84]:
```suggestion
func EqualBaseUrl(url1, url2 string) bool {
```

**Review by iosmanthus — APPROVED** [review]
Rest LGTM

**Review by iosmanthus — APPROVED** [review]

File: pkg/storage/endpoint/tso_keyspace_group.go [hunk 4]
```diff
@@ -80,6 +81,14 @@ type KeyspaceGroupMember struct {
 	Priority int    `json:"priority"`
 }
 
+// CompareAddress compares the address with the given address.
+// It compares the address without the scheme.
+// Otherwise, it will not work when we update the scheme from http to https.
+// Issue: https://github.com/tikv/pd/issues/8284
+func (m *KeyspaceGroupMember) CompareAddress(addr string) bool {
```
Comment (rleungx) [inline pkg/storage/endpoint/tso_keyspace_group.go:88]: Why not directly use EqualBaseURLs?

Comment (lhy1024) [reply pkg/storage/endpoint/tso_keyspace_group.go:88]: Avoid to add comments anywhere, so wrap it.

**Review by rleungx — APPROVED** [review]

**Pull request merged**